code/game/machinery/sprinkler.dm

```diff
@@ -0,0 +1,246 @@
+///basic plumbing machinery it holds a wrench_act used in all plumbing machinery
+/obj/machinery/plumbing
+	name = "pipe thing"
+	icon = 'icons/obj/plumbing/plumbers.dmi'
+	icon_state = "pump"
+	anchored = FALSE
+	active_power_usage = 30
+	use_power = ACTIVE_POWER_USE
+	///how much chems it can hold
+	var/volume = 100
+	///var to prevent do_after stacking
+	var/working = FALSE
+	///if crowbar'd what it turns into
+	var/deployable = null
+
+/obj/machinery/plumbing/Initialize()
+	create_reagents(volume, OPENCONTAINER | AMOUNT_VISIBLE)
+	update_icon()
+	power_change()
+	return ..()
+
+/obj/machinery/plumbing/ComponentInitialize()
+	AddComponent(/datum/component/plumbing/input)
```

Comment:
`. = ..()` at the top

Also this proc is kinda outdated, you can put this in init if you prefer